src/razor/src/dynamicFile/dynamicFileInfoHandler.ts

```diff
@@ -40,7 +40,7 @@ export class DynamicFileInfoHandler {
             }
         );
         this.documentManager.onChange(async (e) => {
-            if (e.kind == RazorDocumentChangeKind.csharpChanged && !e.document.isOpen) {
+            if (e.kind == RazorDocumentChangeKind.csharpChanged && !e.document.isOpen && e.changes.length > 0) {
```

Comment:
Thanks for walking through this with me offline! Since it's perf sensitive, consider adding a comment here to describe the conditions that we *don't* want to update the dynamic file info.